File: union/frontend/src/services/union.service.ts
import api from './api';
import { Union } from '../types';

export const unionService = {
  createUnion: async (data: { name: string; description?: string }): Promise<Union> => {
    const response = await api.post('/unions/createUnion', data);
    return response.data;
  },

  getUserUnions: async (): Promise<Union[]> => {
    try {
      const response = await api.get('/unions/myUnions');
      return response.data || [];
    } catch (error) {
      console.error('Failed to get unions:', error);
      return [];
    }
  },

  getUnionById: async (id: number): Promise<Union> => {
    const response = await api.get(`/unions/getUnionInfo/${id}`);
    return response.data;
  },

  addMember: async (unionId: number, userId: number): Promise<void> => {
    await api.post(`/unions/${unionId}/addUserToUnion/${userId}`);
  },

  removeMember: async (unionId: number, userId: number): Promise<void> => {
    await api.delete(`/unions/${unionId}/removeUserFromUnion/${userId}`);
  },

  updateUnion: async (id: number, data: { name?: string; description?: string }): Promise<Union> => {
    const response = await api.put(`/unions/updateUnionInfo/${id}`, data);
    return response.data;
  },

  deleteUnion: async (id: number): Promise<void> => {
    await api.delete(`/unions/deleteUnion/${id}`);
  },
};

File: union/src/main/java/com/union/app/messaging/service/UnionService.java
package com.union.app.messaging.service;

import com.union.app.user.model.User;
import com.union.app.messaging.dto.UnionCreateDto;
import com.union.app.messaging.model.Union;
import java.util.List;

public interface UnionService {
    Union createUnion(UnionCreateDto dto, User creator);

    List<Union> getUserUnions(User user);

    Union getUnionById(Long id);

    void addMember(Long unionId, Long userId);

    void removeMember(Long unionId, Long userId);

    Union updateUnion(Long id, UnionCreateDto dto);

    void deleteUnion(Long id);

    boolean isMember(Long unionId, User user);
}